sentry-android-core/src/test/java/io/sentry/android/core/AndroidOptionsInitializerTest.kt

```diff
@@ -181,13 +188,47 @@ class AndroidOptionsInitializerTest {
     }
 
     @Test
-    fun `NdkIntegration added to integration list`() {
-        val sentryOptions = SentryAndroidOptions()
-        val mockContext = createMockContext()
+    fun `NdkIntegration availability check won't throw`() {
+        val bundle = Bundle().apply {
+            putString(ManifestMetadataReader.DSN, "https://key@sentry.io/123")
+        }
+        val mockContext = ContextUtilsTest.mockMetaData(metaData = bundle)
+        val logger = mock<ILogger>()
+        val sentryOptions = SentryAndroidOptions().apply {
+            isDebug = true
+        }
+        val buildInfo = mock<IBuildInfoProvider>()
+        whenever(buildInfo.sdkInfoVersion).thenReturn(16)
+
+        // hard to test, lets just check that its not throwing anything
```

Comment:
@bruno-garcia to be able to test this properly, I'd need to have an interface that hides the static call `Class.forName`, so I can mock and assert it, do you think it's worth it?

Reply:
With your suggested refactor of having `loadNdk..` return the class, we could test this by making sure `NdkIntegration` initialized with `null` for `Class<?>` and with an instance it does?

I'd say it's your call. It's nice to have it testable and be able to assert this properly but on the other hand it bloats the SDK further with 1 more type just for this 1 test case.

Reply:
yes this is fully testable/tested now

Reply:
I've added a new type for the sake of testability, this is a critical part of the sdk, as it bundles other sdks and it'd be good to be sure that this is working properly, although those types are package-private anyway.